mautic/mautic
# Extend company fields available in "Send email to user" campaign action

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | staging for features or enhancements / 3.0 for bug fixes 
| Bug fix?                               | yes
| New feature?                           | no
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | yes/no
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 




#### Description:
Only Company Name, City and Country tokens are replaced in emails when using "Send email to user" campaign action.
This PR changes the "Send email to user" campaign action to return all available company fields from the repository so that all of the tokens get replaced.


#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Create a segment
3. Create a company, fill Company Name and "Address 1"
4. Create a contact and set company to previously created company
5. Add the contact to the previously created segment
6. Create an email and add `{contactfield=companyname}` and `{contactfield=companyaddress1}` tokens to the email
7. Create a campagin, choose the previously create segment as start segment
8. Add "Send email to user" campaign action and choose previously created email
9. Run segment rebuild and campaign trigger jobs
10. Check the email and verify if both tokens were replaced correctly
